staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -83,8 +83,9 @@ func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr *generic.Version
 // note that callAttrMutatingHook updates attr
 func (a *mutatingDispatcher) callAttrMutatingHook(ctx context.Context, h *v1beta1.Webhook, attr *generic.VersionedAttributes) error {
 	if attr.IsDryRun() {
-		// TODO: support this
-		return webhookerrors.NewDryRunUnsupportedErr(h.Name)
+		if h.SideEffects == nil || *h.SideEffects == v1beta1.SideEffectClassUnknown || *h.SideEffects == v1beta1.SideEffectClassSome {
```

Comment:
Given the defaulting, it should never be nil, right? It seems like that should be a different kind of error. I guess I'm not asking for a change, just thinking out loud.

Reply:
would it be better to error unless a value of `None` or `NoneOnDryRun` is set? that would let us introduce additional enum values in the future more safely if needed.